Error updating to 1.0.8

When I try to update Rcpp from 1.0.7 to 1.0.8 from source, I get the following errors: 
> install.packages("Rcpp")
clang++ -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fPIC  -Wall -g -O2  -c api.cpp -o api.o
clang++ -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fPIC  -Wall -g -O2  -c attributes.cpp -o attributes.o
attributes.cpp:2813:45: error: a space is required between consecutive right angle brackets (use '> >')
            Rcpp::as<std::vector<std::string>>(pargs_cv);
                                            ^~
                                            > >
1 error generated.
make: *** [attributes.o] Error 1

Output of my sessionInfo(): 
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] MASS_7.3-51.6   compiler_4.0.2  Matrix_1.2-18   tools_4.0.2     parallel_4.0.2  OpenMx_2.18.1   Rcpp_1.0.7      grid_4.0.2     
 [9] digest_0.6.27   packrat_0.5.0   lifecycle_1.0.0 rlang_0.4.11    lattice_0.20-41

Thank you in advance!

@Enchufa If I entertain adding the space _and_ setting C++98 I hit

```sh
edd@rob:~/git/rcpp(master)$ install.r
* installing *source* package found in current working directory ...
* installing *source* package ‘Rcpp’ ...
** using staged installation
** libs
ccache g++-11 -std=c++98 -I"/usr/share/R/include" -DNDEBUG -I../inst/include/     -fpic  -g -O3 -Wall -pipe   -c api.cpp -o api.o
ccache g++-11 -std=c++98 -I"/usr/share/R/include" -DNDEBUG -I../inst/include/     -fpic  -g -O3 -Wall -pipe   -c attributes.cpp -o attributes.o
attributes.cpp: In member function ‘std::string Rcpp::attributes::Attribute::customRSignature() const’:
attributes.cpp:404:20: error: ‘std::string’ {aka ‘class std::__cxx11::basic_string<char>’} has no member named ‘back’
  404 |             if(sig.back() == '}')
      |                    ^~~~
attributes.cpp:408:20: error: ‘std::string’ {aka ‘class std::__cxx11::basic_string<char>’} has no member named ‘front’
  408 |             if(sig.front() == '{')
      |                    ^~~~~
make: *** [/usr/lib/R/etc/Makeconf:177: attributes.o] Error 1
ERROR: compilation failed for package ‘Rcpp’
* removing ‘/usr/local/lib/R/site-library/Rcpp’
* restoring previous ‘/usr/local/lib/R/site-library/Rcpp’
Warning message:
In install.packages(pkgs = f, lib = lib, repos = if (isMatchingFile(f)) NULL else repos) :
  installation of package ‘.’ had non-zero exit status
edd@rob:~/git/rcpp(master)$ 
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2022-01-20  Dirk Eddelbuettel  <edd@debian.org>
+
+	* src/attributes.cpp: Make three small changes to permit compilation
+	under C++98 now (while we consider just turning to C++11 overall)
+
 2022-01-14  Dirk Eddelbuettel  <edd@debian.org>
 
 	* inst/tinytest/test_packageversion.R: Comparison to 'dev' revision

--- a/src/attributes.cpp
+++ b/src/attributes.cpp
@@ -2,7 +2,7 @@
 // attributes.cpp: Rcpp R/C++ interface class library -- Rcpp attributes
 //
 // Copyright (C) 2012 - 2020  JJ Allaire, Dirk Eddelbuettel and Romain Francois
-// Copyright (C) 2021         JJ Allaire, Dirk Eddelbuettel, Romain Francois, Iñaki Ucar and Travers Ching
+// Copyright (C) 2021 - 2022  JJ Allaire, Dirk Eddelbuettel, Romain Francois, Iñaki Ucar and Travers Ching
 //
 // This file is part of Rcpp.
 //
@@ -400,12 +400,20 @@ namespace attributes {
             Param sigParam = paramNamed(kExportSignature);
             std::string sig = sigParam.value();
             trimWhitespace(&sig);
-            if(sig.empty()) return sig;
-            if(sig.back() == '}')
+            if (sig.empty()) return sig;
+#if __cplusplus < 201103L
+            if (sig[sig.size() - 1] == '}')
+#else
+            if (sig.back() == '}')
+#endif
                 sig = sig.substr(0, sig.size()-1);
             // check sig.empty again since we deleted an element
-            if(sig.empty()) return sig;
-            if(sig.front() == '{')
+            if (sig.empty()) return sig;
+#if __cplusplus < 201103L
+            if (sig[0] == '{')
+#else
+            if (sig.front() == '{')
+#endif
                 sig.erase(0,1);
             return sig;
         }
@@ -2810,7 +2818,7 @@ namespace attributes {
         // as the error message is generally more descriptive
         CharacterVector pargs_cv = formalArgs(eval(parse(_["text"] = args)));
         std::vector<std::string> parsed_args =
-            Rcpp::as<std::vector<std::string>>(pargs_cv);
+            Rcpp::as<std::vector<std::string> >(pargs_cv);
 
         for(size_t i=0; i<required_args.size(); ++i) {
             if(std::find(parsed_args.begin(), parsed_args.end(),

